cmd/buildah/manifest.go

```diff
@@ -432,25 +432,21 @@ func manifestAddCmd(c *cobra.Command, args []string, opts manifestAddOpts) error
 	switch len(args) {
 	case 0, 1:
 		return errors.New("At least a list image and an image or artifact to add must be specified")
-	case 2:
+	default:
 		listImageSpec = args[0]
 		if listImageSpec == "" {
 			return fmt.Errorf(`Invalid image name "%s"`, args[0])
 		}
 		if opts.artifact {
 			artifactSpec = args[1:]
 		} else {
-			imageSpec = args[1]
-			if imageSpec == "" {
-				return fmt.Errorf(`Invalid image name "%s"`, args[1])
+			if len(args) != 2 {
+				imageSpec = args[1]
+				if imageSpec == "" {
+					return fmt.Errorf(`Invalid image name "%s"`, args[1])
+				}
 			}
 		}
-	default:
-		if opts.artifact {
-			artifactSpec = args[1:]
-		} else {
-			return errors.New("Too many arguments: expected list and image add to list")
-		}
 	}
```

Comment:
This could have been just:

```
if len(args) < 2 {
  return errors.New("At least a list image and an image or artifact to add must be specified")
}

// the rest of the code - three idents saved
```